test/common/testlib.py

```diff
@@ -460,19 +460,26 @@ def run(self, result=None):
             if startTestRun is not None:
                 startTestRun()
 
-        # Policy actually dictates retries the number here is an upper bound
-        for retry in range(0, 5):
+        self.currentResult = result
+
+        # Here's the loop to actually retry running the test. It's an awkward
+        # place for this loop, since it only applies to MachineCase based
+        # TestCases. However for the time being there is no better place for it.
+        #
+        # Policy actually dictates retries.  The number here is an upper bound to
+        # prevent endless retries if Policy.check_retry is buggy.
+        max_retry_hard_limit = 10
+        for retry in range(0, max_retry_hard_limit):
             try:
-                self.currentResult = result
                 super(MachineCase, self).run(result)
             except RetryError, ex:
-                self.doCleanups()
```

Comment:
It's moved to the TapResult itself which is the main point for managing retries. The retry loop is as clean as possible ... it's unfortunate that the retry loop is here. I wish it could be elsewhere, not in MachineCase specific code. But realistically this is where it has to live for now. At least lets keep it really simple.